### ISSUE TITLE
Template nodes that construct output to grouped media players still fail in 11.0.3.

### ISSUE THREAD

**Issue by grantalewis** [zachowj/node-red-contrib-home-assistant-websocket#591]
### Describe the bug

One of the issues listed in #584 still exists in 11.0.3: template nodes that construct output to grouped media players fail beginning with 11.0.0.

### To Reproduce

See below for example flow that worked in 10.4.0.

### Expected behavior

Same behavior as in Node-RED 10.4.0.

### Example Flow

```json
[{"id":"753287c077f86a13","type":"inject","z":"922c617359cdf6d0","name":"Alexa TTS (no \"ding\")","repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":180,"y":200,"wires":[["3765324511a9ab75"]]},{"id":"3765324511a9ab75","type":"template","z":"922c617359cdf6d0","name":"","field":"payload","fieldType":"msg","format":"handlebars","syntax":"mustache","template":"{\"data\":{\"message\":\"This is a test of TTS with a template node.\",\"data\":{\"type\":\"tts\",\"method\":\"all\"},\"target\":\"group.alexa_wake_downstairs\"}}","output":"str","x":360,"y":200,"wires":[["21dfb3b0c16fe3a9"]]},{"id":"21dfb3b0c16fe3a9","type":"api-call-service","z":"922c617359cdf6d0","name":"Downstairs Wake Group","server":"36944218.4145ce","version":5,"debugenabled":false,"domain":"notify","service":"alexa_media","areaId":[],"deviceId":[],"entityId":[],"data":"","dataType":"json","mergeContext":"","mustacheAltTags":false,"outputProperties":[],"queue":"none","x":550,"y":200,"wires":[[]]},{"id":"36944218.4145ce","type":"server","name":"Home Assistant","version":2,"addon":true,"rejectUnauthorizedCerts":true,"ha_boolean":"y|yes|true|on|home|open","connectionDelay":true,"cacheJson":true,"heartbeat":false,"heartbeatInterval":30}]
```
```


### This package's version (not the Home Assistant add-on version)

0.33.1

### Is Node-RED running in Docker?

Yes

### Node-RED version

1.3.5

### Node.js version

14.17.1

### Additional context

_No response_

**Comment by zachowj** [comment]
The input of a call-service node was never expected to accept the data field as a string. It only worked as a side effect of another property. 

https://zachowj.github.io/node-red-contrib-home-assistant-websocket/node/call-service.html#data-2

If you change the Output As to JSON in the template node. It should start working again.

![image](https://user-images.githubusercontent.com/37859597/154594088-3622c107-cecc-4168-a934-0677bf7a97a5.png)

**Comment by grantalewis** [comment]
Thank you. Verified that you are correct! Very much appreciated.

**Comment by grantalewis** [comment]
Reopening because something is definitely not right.

This test works:
```
[{"id":"17e02d5de9dd982c","type":"inject","z":"922c617359cdf6d0","name":"Announcement (\"ding\") - single","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":290,"y":940,"wires":[["f90d2c9f236264fb"]]},{"id":"f90d2c9f236264fb","type":"api-call-service","z":"922c617359cdf6d0","name":"Family Room","server":"36944218.4145ce","version":5,"debugenabled":false,"domain":"notify","service":"alexa_media","areaId":[],"deviceId":[],"entityId":[],"data":"{\"message\":\"Testing the announce function on the family room echo.\",\"data\":{\"type\":\"announce\",\"method\":\"all\"},\"target\":\"media_player.family_room_echo\"}","dataType":"json","mergeContext":"","mustacheAltTags":false,"outputProperties":[],"queue":"none","x":510,"y":940,"wires":[[]]},{"id":"36944218.4145ce","type":"server","name":"Home Assistant","version":2,"addon":true,"rejectUnauthorizedCerts":true,"ha_boolean":"y|yes|true|on|home|open","connectionDelay":true,"cacheJson":true,"heartbeat":false,"heartbeatInterval":30}]
```
On the very same tab/flow, I copied and pasted the above and in the message changed "family room" to "office." This test gives an API error, "Call-service error. extra keys not allowed @ data['entity_id']."
```
[{"id":"9fc16c48c2a86c2d","type":"inject","z":"922c617359cdf6d0","name":"Announcement (\"ding\") - single","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":290,"y":1000,"wires":[["2e1194d40abaf1fc"]]},{"id":"2e1194d40abaf1fc","type":"api-call-service","z":"922c617359cdf6d0","name":"Family Room","server":"36944218.4145ce","version":5,"debugenabled":false,"domain":"notify","service":"alexa_media","areaId":[],"deviceId":[],"entityId":["light.gavins_lamp"],"data":"{\"message\":\"Testing the announce function on the office echo.\",\"data\":{\"type\":\"announce\",\"method\":\"all\"},\"target\":\"media_player.family_room_echo\"}","dataType":"json","mergeContext":"","mustacheAltTags":false,"outputProperties":[],"queue":"none","x":510,"y":1000,"wires":[[]]},{"id":"36944218.4145ce","type":"server","name":"Home Assistant","version":2,"addon":true,"rejectUnauthorizedCerts":true,"ha_boolean":"y|yes|true|on|home|open","connectionDelay":true,"cacheJson":true,"heartbeat":false,"heartbeatInterval":30}]
```

<img width="416" alt="Screen Shot 2022-02-19 at 4 43 20 PM" src="https://user-images.githubusercontent.com/9505044/154820154-f9973259-9a37-49f4-b8b2-740f6f6ca36c.png">

**Comment by zachowj** [comment]
Guessing the notify.alexa_media doesn't accept an entity id

![image](https://user-images.githubusercontent.com/37859597/154820287-f1290f8f-5923-4b85-92a9-4026263e57ae.png)

**Comment by grantalewis** [comment]
How are you reaching that view that shows the **light.gavins_lamp**? (I updated with additional screen shots above -- that light isn't anywhere I can see.)

**Comment by zachowj** [comment]
It's because I don't have the alexa_media components so it shows all the target fields. If you switch to a domain/service that shows all the targets you'll see it, `homeassistant`/`toggle`.

The latest release, v0.42.7, will remove any areas, devices, and entities if the service doesn't need them.

**Comment by grantalewis** [comment]
Thank you for the reply. Unfortunately I don't follow what you're saying.

Did two more tests, this time building from scratch rather than copying something that existed previously. Both were successful.

Announce test:
```
[{"id":"41486f800719eda7","type":"inject","z":"922c617359cdf6d0","name":"Announce Office","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":260,"y":1080,"wires":[["a9813de11bc65aa9"]]},{"id":"a9813de11bc65aa9","type":"api-call-service","z":"922c617359cdf6d0","name":"Office","server":"36944218.4145ce","version":5,"debugenabled":false,"domain":"notify","service":"alexa_media","areaId":[],"deviceId":[],"entityId":[],"data":"{\"message\":\"Testing the announce function on the office echo.\",\"data\":{\"type\":\"announce\",\"method\":\"all\"},\"target\":\"media_player.office_echo\"}","dataType":"json","mergeContext":"","mustacheAltTags":false,"outputProperties":[],"queue":"none","x":410,"y":1080,"wires":[[]]},{"id":"36944218.4145ce","type":"server","name":"Home Assistant","version":2,"addon":true,"rejectUnauthorizedCerts":true,"ha_boolean":"y|yes|true|on|home|open","connectionDelay":true,"cacheJson":true,"heartbeat":false,"heartbeatInterval":30}]
````
TTS test:
```
[{"id":"1585fa493fee6008","type":"inject","z":"922c617359cdf6d0","name":"TTS Office","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":240,"y":1140,"wires":[["871540e1dd937b2b"]]},{"id":"871540e1dd937b2b","type":"api-call-service","z":"922c617359cdf6d0","name":"Office","server":"36944218.4145ce","version":5,"debugenabled":false,"domain":"notify","service":"alexa_media","areaId":[],"deviceId":[],"entityId":[],"data":"{\"message\":\"Testing the tts function on the office echo.\",\"data\":{\"type\":\"tts\",\"method\":\"all\"},\"target\":\"media_player.office_echo\"}","dataType":"json","mergeContext":"","mustacheAltTags":false,"outputProperties":[],"queue":"none","x":410,"y":1140,"wires":[[]]},{"id":"36944218.4145ce","type":"server","name":"Home Assistant","version":2,"addon":true,"rejectUnauthorizedCerts":true,"ha_boolean":"y|yes|true|on|home|open","connectionDelay":true,"cacheJson":true,"heartbeat":false,"heartbeatInterval":30}]
```
So this definitely seems like a bug, a departure from prior behavior. What's the next step to a resolution so that copying and pasting works as it did previously?

**Comment by zachowj** [comment]
I believe the issue you are seeing is fixed in the latest release. Are you running the latest version?  v0.42.7

**Comment by grantalewis** [comment]
Perhaps not. I'm running Node-RED through Home Assistant and have the most current available version, 11.0.3. But when I look at the changelog, I see:

- Bump node-red from 2.2.0 to 2.2.1 in /node-red @dependabot (#1294)
- Bump node-red-contrib-home-assistant-websocket from 0.42.4 to 0.42.5 in /node-red @dependabot (#1298)
